grafana/grafana
# InfluxDB: backend migration (run query in explore)

**What this PR does / why we need it**:
Part of https://github.com/grafana/grafana/issues/43076

This PR contains the changes needed to run InfluxQL query requests (that are built in Explore) through the backend.

**Test scenarios**:

**Explore** via Server access mode

1. Run make devenv sources=influxdb,influxdb1,grafana grafana_version=8.2.4 (or other version before changes)
2. Open up http://localhost:3000/explore && http://127.0.0.1:3001/explore (to test older 8.2.4 version)
3. Select gdev-influxdb-influxql
    - Getting dropdown values for from, where, select, group by should query `api/datasources/proxy/38/query`
    - Selecting dropdown values  for from, where, select, group by should query `api/ds/query`
    - Setting values for time zone, order by time, limit, slimit, format as should query `api/ds/query`
    - Raw query mode should query `api/ds/query`
    - `Inspector -> Query` should show correct query
    - Changing the above widget values should reflect the data shown
    - Multiple queries should behave as expected
4. Select gdev-influxdb-flux
    - Queries should go through `api/ds/query`    
5. Select gdev-influxdb1-influxql
    - Should behave like `gdev-influxdb-influxql`
